Add tutorial for client tools (#37)
fixie-ai/ultradox: docs/src/content/docs/guides/clienttoolstutorial.mdx

We'll build a drive-thru order display for a fictional restaurant called "Dr. Donut". The display will update in real-time as customers place orders through our AI agent.

### Implementation Steps
<Steps>
1. **Define the Tool** → Create a schema for order updates
1. **Implement Logic** → Build the tool's functionality
1. **Register the Tool** → Connect it to the Ultravox system
1. **Create the UI** → Build the order display component
</Steps>

:::tip[Stuck?]
If at any point you get lost, you can refer to the `/final` folder in the repo to get final versions of the various files you will create or edit.
:::

## Step 1: Define the Tool

First, we'll define our `updateOrder` tool that the AI agent will use to modify the order display.

Modify `.demo-config.ts`:

```ts
const selectedTools: SelectedTool[] = [
  {
    "temporaryTool": {
      "modelToolName": "updateOrder",
      "description": "Update order details. Used any time items are added or removed or when the order is finalized. Call this any time the user updates their order.",      
      "dynamicParameters": [
        {
          "name": "orderDetailsData",
          "location": ParameterLocation.BODY,

>           "schema": {

Reviewer: If this schema is working well for you then great, but I'd expect better model adherence by splitting this into {add,update,remove}OrderedItem so that your nested parameters can be top-level.

>             "description": "An array of objects contain order items.",

Reviewer: ```suggestion
            "description": "An array of objects representing order items.",
```

            "type": "array",
            "items": {
              "type": "object",
              "properties": {

>                 "name": { "type": "string", "description": "The name of the item to be added to the order." },

Reviewer: ```suggestion
                "name": { "type": "string", "description": "The name of the ordered menu item." },
```
